### PR TITLE
Order list has been added.

### PR DESCRIPTION
**Pull request by l0lka**

### CODE REVIEW

**Comment by sergeyamator** [comment]
Please, resolve conflicts 
src/common/reducers/index.js
src/config/navigator.js

File: src/components/main-order-list/reducers/reducers.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import GET_ORDERS from '../constants';
+
+const initialState = [];
+
+export default (state = initialState, action) => {
```
Comment (sergeyamator) [inline src/components/main-order-list/reducers/reducers.js]: How about tests ?

File: src/components/main-order-list/actions/orderActions.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+const getOrdersUrl = 'https://private-bf0eb-test12906.apiary-mock.com/orders';
+
+export const getOrders = (data) => {
+  return {
+    type: GET_ORDERS,
+    payload: data,
+  };
+};
+
+export const loadOrders = () => {
```
Comment (sergeyamator) [inline src/components/main-order-list/actions/orderActions.js]: How about tests ? 
Where error action and request action ?

File: .babelrc [hunk 3]
```diff
@@ -1,7 +1,10 @@
 {
   "presets": [
     "babel-preset-react-native-stage-0/decorator-support",
-    "jest"
+    "jest",
+    "react-native",
+    "es2015",
+    "babel-preset-stage-3"
```
Comment (serhiirubets) [inline .babelrc]: Are you sure you need all this changes ?

File: package.json [hunk 4]
```diff
@@ -31,7 +32,11 @@
     "preset": "react-native"
   },
   "dependencies": {
+    "asyncawait": "^1.0.6",
+    "babel-preset-es2015": "^6.24.1",
+    "babel-preset-stage-2": "^6.24.1",
     "eslint-plugin-react-native": "^3.2.1",
+    "expect": "^22.1.0",
```
Comment (serhiirubets) [inline package.json]: I don't think we nedd this changes in packege json

**Comment by serhiirubets** [comment]
You have conflicts 
src/common/reducers/index.js
src/config/navigator.js

File: package.json [hunk 5]
```diff
@@ -38,6 +38,7 @@
   },
   "dependencies": {
     "eslint-plugin-react-native": "^3.2.1",
+    "expect": "^22.1.0",
```
Comment (serhiirubets) [inline package.json]: Are you sure you need expect ?

File: .babelrc [hunk 6]
```diff
@@ -10,4 +10,4 @@
       ]
     }
   }
-}
\ No newline at end of file
+}
```
Comment (serhiirubets) [inline .babelrc]: extra new line

**Comment by serhiirubets** [comment]
Please, take latest changes from master, it helps to resolve eslint errors

File: src/components/order-item/actions/order-detais-action.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+};
+
+export const loadingFailed = (payload) => {
+  return {
+    type: LOADING_FAILED,
+    payload,
+  };
+};
+
+
```
Comment (sergeyamator) [inline src/components/order-item/actions/order-detais-action.js]: Extra line

File: src/components/order-item/detailed-order.js [hunk 8]
```diff
@@ -1,56 +1,117 @@
...
+      total: PropTypes.string,
+      deliveryStatus: PropTypes.string,
+      feedback: PropTypes.shape({
+        status: PropTypes.string,
+        statusCode: PropTypes.string,
+        comment: PropTypes.string,
+      }),
+    }),
+    loadOrderDetails: PropTypes.func,
+  };
```
Comment (sergeyamator) [inline src/components/order-item/detailed-order.js]:
```
state = {
 selectedIndex: 2,
}
```
Pls, remove constructor

File: src/components/order-list/actions/order-actions.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+export const loadOrders = () => {
+  return (dispatch) => {
+    return fetch(getOrdersUrl, {
+      method: 'GET',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+    })
+      .then((response) => {
```
Comment (sergeyamator) [inline src/components/order-list/actions/order-actions.js]: ` .then((response) =>  response.json();`

File: src/components/order-list/actions/order-actions.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+        return response.json();
+      })
+      .then((data) => {
```
Comment (sergeyamator) [inline src/components/order-list/actions/order-actions.js]: ` .then((data) =>  dispatch(getOrders(data));`

File: src/components/order-list/order-item.js [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+import { bindActionCreators } from 'redux';
+import { getOrderDetails } from './actions/order-actions';
+
+import Button from '../button';
+import ProductItem from './product-item';
+
+import styles from './styles';
+
+@connect(state => state, dispatch => bindActionCreators({ getOrderDetails }, dispatch))
+
```
Comment (sergeyamator) [inline src/components/order-list/order-item.js]: Extra line

File: src/components/order-list/order-list.js [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+import { bindActionCreators } from 'redux';
+import PropTypes from 'prop-types';
+
+import styles from './styles';
+import { loadOrders } from './actions/order-actions';
+import OrderItem from './order-item';
+import SortForm from '../sort-form';
+import Loader from '../loader';
+
+
```
Comment (sergeyamator) [inline src/components/order-list/order-list.js]: Extra line

File: src/components/sort-form/sort-form.js [hunk 13]
```diff
@@ -19,33 +19,28 @@ function mapDispatchToProps() {
 @connect(mapStateToProps, mapDispatchToProps)
 export default class SortForm extends PureComponent {
   static propTypes = {};
+  constructor() {
+    super();
+    this.state = {
+      selectedIndex: 0,
+    };
+  }
```
Comment (sergeyamator) [inline src/components/sort-form/sort-form.js]:
```
constructor() {
    super();
    this.state = {
     selectedIndex: 0,
    };
 }
```

Will be better to change for this
```
state = {
   selectedIndex: 0,
}
```

File: src/components/feedback/feedback.js [hunk 14]
```diff
@@ -0,0 +1,98 @@
+import React, { PureComponent } from 'react';
+import { ScrollView, View, Text, TouchableOpacity } from 'react-native';
+import FontAwesome, { Icons } from 'react-native-fontawesome';
+import { Button, Icon } from 'react-native-elements';
+import { AutoGrowingTextInput } from 'react-native-autogrow-textinput';
+import styles from './styles';
+
+export default class Loader extends PureComponent {
+  constructor(props) {
```
Comment (sergeyamator) [inline src/components/feedback/feedback.js]: Please, remove constructor

File: src/components/order-item/actions/order-detais-action.js [hunk 15]
```diff
@@ -0,0 +1,39 @@
+import { GET_ORDER_DETAILS, LOADING_FAILED } from '../constants';
+
+const getOrderDetailsUrl =
+  'https://private-bf0eb-test12906.apiary-mock.com/order';
+
+export const getOrderDetails = (data) => {
+  console.log('...Details order... ', data);
```
Comment (sergeyamator) [inline src/components/order-item/actions/order-detais-action.js]: Pls, remove console log

File: src/components/loader/loader.js [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+
+export default class Loader extends PureComponent {
+  render() {
+    return (
+      <View style={styles.container}>
+        <Image
+          style={{ width: 50, height: 50 }}
+          source={{
+            uri:
+              'https://loading.io/spinners/coolors/lg.palette-rotating-ring-loader.gif',
```
Comment (httpEugene) [inline src/components/loader/loader.js]: Why it on new line?

File: src/components/order-item/actions/order-detais-action.js [hunk 17]
```diff
@@ -0,0 +1,33 @@
+import { GET_ORDER_DETAILS, LOADING_FAILED } from '../constants';
+
+const getOrderDetailsUrl =
+  'https://private-bf0eb-test12906.apiary-mock.com/order';
```
Comment (httpEugene) [inline src/components/order-item/actions/order-detais-action.js]: Why it on new line?

File: src/components/order-item/actions/order-detais-action.js [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+export const loadingFailed = (payload) => {
+  return {
+    type: LOADING_FAILED,
+    payload,
+  };
+};
+
+export const loadOrderDetails = (id) => {
+  return (dispatch) => {
+    return fetch(`${getOrderDetailsUrl}/${id}`, {
```
Comment (httpEugene) [inline src/components/order-item/actions/order-detais-action.js]: Consider to use ApiService in future

File: src/components/order-item/reducers/order-details-reducer.js [hunk 19]
```diff
@@ -0,0 +1,14 @@
+import { GET_ORDER_DETAILS } from '../constants';
+
+const initialState = {};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
```
Comment (httpEugene) [inline src/components/order-item/reducers/order-details-reducer.js]: It seems that switch is redundant here

**Pull request merged**